### PR TITLE
Document network support for configdrive

### PR DESCRIPTION
**Pull request by vishesh92**
To be merged after https://github.com/apache/cloudstack/pull/9329

----
📚 Documentation preview 📚: https://cloudstack-documentation--412.org.readthedocs.build/en/412/

### CODE REVIEW

File: source/adminguide/networking/dns_and_dhcp.rst [hunk 1]
```diff
@@ -17,6 +17,13 @@
...
-proxies DNS requests to the DNS server configured on the Availability
-Zone.
+The Virtual Router & ConfigDrive (since Apache CloudStack 4.20) provides
+DNS and DHCP services to the guests. It proxies DNS requests to the DNS
+server configured on the Availability Zone.
+
+.. note::
+   In case of a network with ConfigDrive, adding/removing nic to/from an
+   instance or updating the ip address of a nic will not be reflected in the
+   instance if the instance is already running. To do so, run
```
Comment (weizhouapache) [inline source/adminguide/networking/dns_and_dhcp.rst:27]: @vishesh92 
if the instance is stopped, will the ip address be updated when it is started ?

**Review by harikrishna-patnala — COMMENTED** [review]
Question (asked the same on the PR also), how this needs to be configured in the network offering ? like we choose the config drive as the provider for userdata while creating the network offering. Or we don't need to ??

**Comment by harikrishna-patnala** [comment]
@blueorangutan docbuild

File: source/adminguide/networking/dns_and_dhcp.rst (hunk 1 above)
Comment (vishesh92) [reply source/adminguide/networking/dns_and_dhcp.rst:27]: It won't as of now. Let me make it more specific here.

**Comment by vishesh92** [comment]
> Question (asked the same on the PR also), how this needs to be configured in the network offering ? like we choose the config drive as the provider for userdata while creating the network offering. Or we don't need to ??

Yes. It's part of the network offering. Earlier, `ConfigDrive` wasn't available as an option for DHCP & DNS. Now it is.
image

image

File: source/adminguide/networking/dns_and_dhcp.rst (hunk 1 above)
Comment (vishesh92) [reply source/adminguide/networking/dns_and_dhcp.rst:27]: This is a limitation of ConfigDrive with cloud-init. It applies network configuration from ConfigDrive only during the first boot.

**Review by harikrishna-patnala — APPROVED** [review]
doc LGTM

File: source/adminguide/networking/dns_and_dhcp.rst (hunk 1 above)
Comment (vladimirpetrov) [reply source/adminguide/networking/dns_and_dhcp.rst:27]: I was definitely able to see the changes when adding/removing NICs only by executing stop/start VM sequence (not restart).

**Pull request merged**